src/core/Akka.Remote/Serialization/LruBoundedCache.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Akka.Remote.Serialization
+{
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <remarks>
+    /// Fast hash based on the 128 bit Xorshift128+ PRNG. Mixes in character bits into the random generator state.
+    /// </remarks>
+    public static class FastHash
+    {
+        public static int OfString(string s)
+        {
+            var chars = s.ToCharArray();
+            var s0 = 391408L;
+            var s1 = 601258L;
+            var i = 0;
+            unchecked
+            {
+                while (i < chars.Length)
```

Comment:
The for syntax was just an easier way to express the pointer arithmetic I needed to traverse the character array and I wrote that function second.